fili-core/src/main/java/com/yahoo/bard/webservice/async/workflows/DefaultAsynchronousWorkflowsBuilder.java

```diff
@@ -204,8 +206,8 @@ public AsynchronousWorkflows buildAsynchronousWorkflows(
         // The job status should not be updated until both the PreResponse has been stored, and the storage of the
         // original Job ticket has been attempted.
         return preResponseEmitter
-                .delay(ignored -> preResponseStoredNotification)
-                .delay(ignored -> jobRowStoredNotification)
+                .zipWith(jobRowStoredNotification, (preResponse, ignored) -> preResponse)
+                .zipWith(preResponseStoredNotification, (preResponse, ignored) -> preResponse)
```

Comment:
The comment above the stream is supposed to explain it, by explaining that we need to wait until the job row has been stored, and the pre response has been stored. I can make it more explicit that we're using the zipWith as a gating mechanism if you'd like.